src/ngraph/graph_util.cpp

```diff
@@ -131,13 +132,21 @@ NodeVector ngraph::find_common_args(std::shared_ptr<Node> node1, std::shared_ptr
     return common_args;
 }
 
-void ngraph::replace_node(std::shared_ptr<Node> target, std::shared_ptr<Node> replacement)
+void ngraph::replace_node(std::shared_ptr<Node> target,
```

Comment:
If this was `replace_value(Output<Node> target, Output<Node> replacement)` it wouldn't need these output size checks and `output_order` arguments.

Reply:
In current implementation: `replace_node(std::shared_ptr<Node> target, std::shared_ptr<Node> replacement)` (line: 198) call extended `replace_node(std::shared_ptr<Node> target, std::shared_ptr<Node> replacement, const std::vector<int64_t>& output_order)` with default `output_order` value (**to avoid copying code**), but you are right that for version without `output_order` it cases additional checking and vector allocation. 
@diyessi  Do you expect fully separate version of `replace_node` for both cases?

Reply:
Let's keep this one for now and think about how we can simplify replace node in general. You've had a fair amount of experience with it now.